flake.nix

```diff
@@ -1,50 +1,290 @@
 {
   description = "cardano-ledger";
 
+  inputs = {
+    haskellNix.url = "github:input-output-hk/haskell.nix";
+    nixpkgs.follows = "haskellNix/nixpkgs-unstable";
+    iohkNix.url = "github:input-output-hk/iohk-nix";
+    flake-utils.url = "github:hamishmack/flake-utils/hkm/nested-hydraJobs";
+
+    CHaP.url = "github:input-output-hk/cardano-haskell-packages?ref=repo";
+    CHaP.flake = false;
+
+    cardano-mainnet-mirror.url = "github:input-output-hk/cardano-mainnet-mirror";
+    cardano-mainnet-mirror.flake = false;
+
+    # cicero
+    tullia.url = "github:input-output-hk/tullia";
+
+    # non-flake nix compatibility
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
+  };
+
+  outputs = inputs: let
+    supportedSystems = [
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-linux"
+      "aarch64-darwin"
+    ];
+  in
+    inputs.flake-utils.lib.eachSystem supportedSystems (
+      system: let
+        # setup our nixpkgs with the haskell.nix overlays, and the iohk-nix
+        # overlays...
+        nixpkgs = import inputs.nixpkgs {
+          overlays = [
+            # our local packages.
+            (import ./nix/pkgs.nix)
+            # iohkNix.overlays.crypto provide libsodium-vrf, libblst and libsecp256k1.
+            inputs.iohkNix.overlays.crypto
+            # haskellNix.overlay can be configured by later overlays, so need to come before them.
+            inputs.haskellNix.overlay
+            # configure haskell.nix to use iohk-nix crypto librairies.
+            inputs.iohkNix.overlays.haskell-nix-crypto
+          ];
+          inherit system;
+          inherit (inputs.haskellNix) config;
+        };
+        inherit (nixpkgs) lib;
+
+        # We use cabalProject' to ensure we don't build the plan for
+        # all systems.
+        cabalProject = nixpkgs.haskell-nix.cabalProject' {
+          src = ./.;
+          name = "cardano-ledger";
+          compiler-nix-name = lib.mkDefault "ghc928";
+
+          # CHaP input map, so we can find CHaP packages (needs to be more
+          # recent than the index-state we set!). Can be updated with
+          #
+          #  nix flake lock --update-input CHaP
+          #
+          inputMap = {
+            "https://input-output-hk.github.io/cardano-haskell-packages" = inputs.CHaP;
+          };
+
+          # tools we want in our shell, from hackage
+          shell.tools = {
+            cabal = "3.10.1.0";
+            ghcid = "0.8.8";
+            haskell-language-server = "latest";
+            hlint = {};
+            fourmolu = "0.12.0.0";
```

Comment:
Maybe it would be better to keep in this PR the version of formolu as it was (`0.10.1.0`), because this new version seems to trigger some reformattings on the current code. Perhaps better if I do this separately, both the upgrade and the reformatting?